### PR TITLE
W-12220317: [Tracer Performance] Remove commands for coreEventTracer or do not generate instances for each request

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

**Comment by fsgonz** [comment]
--validate

File: modules/tracing/mule-tracer-internal-impl/src/main/java/org/mule/runtime/tracer/impl/span/command/EventContextAddAttributesCommand.java [hunk 1]
```diff
@@ -7,47 +7,40 @@
...
+import org.slf4j.Logger;
 
 import java.util.Map;
-
-import static org.mule.runtime.tracer.impl.span.command.spancontext.SpanContextFromEventContextGetter.getSpanContextFromEventContextGetter;
+import java.util.function.BiConsumer;
 
 /**
- * A {@link VoidCommand} that ads span attributes. The carrier is the {@link EventContext}
+ * A {@link AbstractFailSafeVoidBiCommand} that ads span attributes. The carrier is the {@link EventContext}
```
Comment (sofiamorseletto) [inline modules/tracing/mule-tracer-internal-impl/src/main/java/org/mule/runtime/tracer/impl/span/command/EventContextAddAttributesCommand.java:20]: adds*

File: modules/tracing/mule-tracer-internal-impl/src/main/java/org/mule/runtime/tracer/impl/span/command/EventContextEndSpanCommand.java [hunk 2]
```diff
@@ -11,55 +11,38 @@
...
-    this.eventContext = eventContext;
-    this.assertion = assertion;
+  private EventContextEndSpanCommand(Logger logger, String errorMessage, boolean propagateExceptions) {
+    super(logger, errorMessage, propagateExceptions);
   }
 
-  protected Runnable getRunnable() {
-    return () -> {
+  @Override
+  BiConsumer<EventContext, Assertion> getConsumer() {
```
Comment (sofiamorseletto) [inline modules/tracing/mule-tracer-internal-impl/src/main/java/org/mule/runtime/tracer/impl/span/command/EventContextEndSpanCommand.java:39]: To understand a little bit more, on the "getFunction" classes such as AbstractFailsafeUnaryCommand, the command is created on the constructor, saved on a variable and then returned on the getter. Why are we not doing the same for the "getConsumer" classes such as this one? Why is the consumer created on the getter and not on the constructor?

Comment (fsgonz) [reply modules/tracing/mule-tracer-internal-impl/src/main/java/org/mule/runtime/tracer/impl/span/command/EventContextEndSpanCommand.java:39]: Yes, you're right. My mistake. It should be created in the constructor.

File: modules/tracing/mule-tracer-internal-impl/src/main/java/org/mule/runtime/tracer/impl/span/command/EventContextInjectDistributedTraceContextCommand.java [hunk 3]
```diff
@@ -13,32 +13,33 @@
...
-    this.eventContext = eventContext;
-    this.getter = getter;
+  private EventContextInjectDistributedTraceContextCommand(Logger logger, String errorMessage, boolean propagateExceptions) {
+    super(logger, errorMessage, propagateExceptions);
   }
 
-  protected Runnable getRunnable() {
-    return () -> {
+  @Override
+  BiConsumer<EventContext, DistributedTraceContextGetter> getConsumer() {
```
Comment (sofiamorseletto) [inline modules/tracing/mule-tracer-internal-impl/src/main/java/org/mule/runtime/tracer/impl/span/command/EventContextInjectDistributedTraceContextCommand.java:41]: According to what we discussed before, the creation of the consumer should be moved to the constructor of the class. This goes for all classes that use the getConsumer method, I assume are the ones that extend `AbstractFailSafeVoidBiCommand`

File: modules/tracing/mule-tracer-internal-impl/src/main/java/org/mule/runtime/tracer/impl/span/command/EventContextAddAttributesCommand.java (hunk 1 above)
Comment (sofiamorseletto) [reply modules/tracing/mule-tracer-internal-impl/src/main/java/org/mule/runtime/tracer/impl/span/command/EventContextAddAttributesCommand.java]: @fsgonz if you can please change this typo too

**Review by sofiamorseletto — CHANGES_REQUESTED** [review]

**Review by sofiamorseletto — APPROVED** [review]

**Pull request merged**